assets/src/dashboard/app/views/myStories/content/storiesView/index.js

```diff
@@ -220,38 +223,80 @@ function StoriesView({
     };
   }, [handleOnRenameStory, setTitleRenameId, titleRenameId]);
 
-  const ActiveView =
-    view.style === VIEW_STYLE.LIST ? (
-      <StoryListView
-        handleSortChange={sort.set}
-        handleSortDirectionChange={sort.setDirection}
-        pageSize={view.pageSize}
-        renameStory={renameStory}
-        sortDirection={sort.direction}
-        stories={stories}
-        storyMenu={storyMenu}
-        storySort={sort.value}
-        storyStatus={filterValue}
-      />
-    ) : (
-      <StoryGridView
-        bottomActionLabel={__('Open in editor', 'web-stories')}
-        centerActionLabelByStatus={
-          enableStoryPreviews && STORY_ITEM_CENTER_ACTION_LABELS
-        }
-        pageSize={view.pageSize}
-        renameStory={renameStory}
-        previewStory={storyActions.handlePreviewStory}
-        storyMenu={storyMenu}
-        stories={stories}
-        returnStoryFocusId={returnStoryFocusId}
-        initialFocusStoryId={initialFocusStoryId}
-      />
-    );
+  const ActiveView = useMemo(() => {
+    // Stories should be shown when we trigger a fetch from `InfiniteScroll`.
+    // Stories should be hidden when a filter is changed.
+    if (view.style === VIEW_STYLE.LIST) {
+      // StoryListView needs to show the table header when loading stories
+      // when filtering.
+      return (
+        <StoryListView
+          handleSortChange={sort.set}
+          handleSortDirectionChange={sort.setDirection}
+          hideStoryList={
+            loading?.isLoading && !loading?.showStoriesWhileLoading.current
+          }
+          pageSize={view.pageSize}
+          renameStory={renameStory}
+          sortDirection={sort.direction}
+          stories={stories}
+          storyMenu={storyMenu}
+          storySort={sort.value}
+          storyStatus={filterValue}
+        />
+      );
+    }
+
+    if (
+      !loading?.isLoading ||
+      (loading?.isLoading && loading?.showStoriesWhileLoading.current)
+    ) {
+      return (
+        <StoryGridView
+          bottomActionLabel={__('Open in editor', 'web-stories')}
+          centerActionLabelByStatus={
+            enableStoryPreviews && STORY_ITEM_CENTER_ACTION_LABELS
+          }
+          isLoading={loading?.isLoading}
+          pageSize={view.pageSize}
+          renameStory={renameStory}
+          previewStory={storyActions.handlePreviewStory}
+          storyMenu={storyMenu}
+          stories={stories}
+          returnStoryFocusId={returnStoryFocusId}
+          initialFocusStoryId={initialFocusStoryId}
+        />
+      );
+    }
+
+    // Hide all stories when filter is triggered.
+    return null;
+  }, [
+    enableStoryPreviews,
+    loading,
+    filterValue,
+    initialFocusStoryId,
+    renameStory,
+    returnStoryFocusId,
+    sort?.direction,
+    sort?.set,
+    sort?.setDirection,
+    sort?.value,
```

Comment:
just curious - what's the benefit here of checking for keys individually instead of just `sort`? - think this is everything in that object.

Reply:
you're right, we don't need to do this